contributors/design-proposals/deploy.md

```diff
@@ -118,29 +118,29 @@ See issue [#17164](https://github.com/kubernetes/kubernetes/issues/17164).
 
-Note that a deployment's replication controllers use PodTemplate hashes (i.e. the hash of `.spec.template`) to distinguish from each others. When doing rollout or rollback, a deployment reuses existing replication controller if it has the same PodTemplate, and its `rollout.kubectl.kubernetes.io/change-source` and `rollout.kubectl.kubernetes.io/version` annotations will be updated by the new rollout. At this point, the earlier state of this replication controller is lost in history. For example, if we had 3 replication controllers in
-deployment history, and then we do a rollout with the same PodTemplate as version 1, then version 1 is lost and becomes version 4 after the rollout.
+Note that a Deployment's ReplicaSets use PodTemplate hashes (i.e. the hash of `.spec.template`) to distinguish from each others. When doing rollout or rollback, a Deployment reuses existing ReplicaSet if it has the same PodTemplate, and its `kubernetes.io/change-cause` and `deployment.kubernetes.io/revision` annotations will be updated by the new rollout. At this point, the earlier state of this ReplicaSet is lost in history. For example, if we had 3 ReplicaSets in
+Deployment history, and then we do a rollout with the same PodTemplate as version 1, then version 1 is lost and becomes version 4 after the rollout.
 
-To make deployment versions more meaningful and readable for the users, we can add more annotations in the future. For example, we can add the following flags to `kubectl` for the users to describe and record their current rollout:
+To make Deployment versions more meaningful and readable for the users, we can add more annotations in the future. For example, we can add the following flags to `kubectl` for the users to describe and record their current rollout:
 - `--description`: adds `description` annotation to an object when it's created to describe the object.
 - `--note`: adds `note` annotation to an object when it's updated to record the change.
 - `--commit`: adds `commit` annotation to an object with the commit id.
 
 ### Pause Deployments
 
-Users sometimes need to temporarily disable a deployment. See issue [#14516](https://github.com/kubernetes/kubernetes/issues/14516).
+Users sometimes need to temporarily disable a Deployment. See issue [#14516](https://github.com/kubernetes/kubernetes/issues/14516).
 
 ### Perm-failed Deployments
 
-The deployment could be marked as "permanently failed" for a given spec hash so that the system won't continue thrashing on a doomed deployment. The users can retry a failed deployment with `kubectl rollout retry`. See issue [#14519](https://github.com/kubernetes/kubernetes/issues/14519).
+The Deployment could be marked as "permanently failed" for a given spec hash so that the system won't continue thrashing on a doomed Deployment. The users can retry a failed Deployment with `kubectl rollout retry`. See issue [#14519](https://github.com/kubernetes/kubernetes/issues/14519).
```

Comment:
I suggest repurposing this section to mention progressDeadlineSeconds and maybe point to our docs about it. Also `kubectl rollout retry` is not a thing so I would remove it from here to avoid any confusion.